src/runtime/crt/host/microtvm_api_server.py

```diff
@@ -67,6 +88,29 @@ def server_info_query(self, tvm_version):
     # The build target given to make
     BUILD_TARGET = "build/main"
 
+    def _populate_makefile(
+        self,
+        makefile_template_path: pathlib.Path,
+        makefile_path: pathlib.Path,
+        memory_size: int,
+    ):
+        """Generate Makefile from template."""
+        flags = {
+            "MEMORY_SIZE_BYTES": str(memory_size),
+        }
+
+        with open(makefile_path, "w") as makefile_f:
+            with open(makefile_template_path, "r") as makefile_template_f:
+                for line in makefile_template_f:
+                    SUBST_TOKEN_RE = re.compile(r"<([A-Z_]+)>")
+                    outs = []
+                    for i, m in enumerate(re.split(SUBST_TOKEN_RE, line)):
+                        if i % 2 == 1:
+                            m = flags[m]
+                        outs.append(m)
+                    line = "".join(outs)
+                    makefile_f.write(line)
```

Comment:
Could we make it shorter with (or something alike to):

```
token_val = {
    "MEMORY": str(memory_size),
}

regex = re.compile(r'([A-Z_]+) := (<[A-Z_]+>)')
with open(makefile_path, "w") as makefile_f:
    with open(makefile_template_path, "r") as makefile_template_f:
        for line in makefile_template_f:
            m = regex.match(line)
            if m:
                var, token = m.groups()
                line = re.sub(token, token_val[var], line)

            makefile_f.write(line)
```
?

Reply:
I changed it to this